frontend/src/app/main/ui/shapes/fills.cljs

```diff
@@ -62,32 +70,50 @@
                       (obj/set! pat-props "patternTransform" transform)
                       pat-props)]
 
-    (for [[shape-index shape] (d/enumerate (or (:position-data shape) [shape]))]
-      [:* {:key (dm/str shape-index)}
-       (for [[fill-index value] (reverse (d/enumerate (get shape :fills [])))]
+    (for [[obj-index obj] (d/enumerate (or (:position-data shape) [shape]))]
+      [:* {:key (dm/str obj-index)}
+       (for [[fill-index value] (reverse (d/enumerate (get obj :fills [])))]
          (when (some? (:fill-color-gradient value))
            (let [gradient  (:fill-color-gradient value)
+
+                 from-p (-> (gpt/point (+ x (* width (:start-x gradient)))
+                                       (+ y (* height (:start-y gradient)))))
+                 to-p   (-> (gpt/point (+ x (* width (:end-x gradient)))
+                                       (+ y (* height (:end-y gradient)))))
+
+                 gradient
+                 (cond-> gradient
+                   (some? bounds)
+                   (assoc
+                    :start-x (/ (- (:x from-p) (:x bounds)) (:width bounds))
+                    :start-y (/ (- (:y from-p) (:y bounds)) (:height bounds))
+                    :end-x   (/ (- (:x to-p) (:x bounds)) (:width bounds))
+                    :end-y   (/ (- (:y to-p) (:y bounds)) (:height bounds))))
+
                  props #js {:id (dm/str "fill-color-gradient-" render-id "-" fill-index)
                             :key (dm/str fill-index)
                             :gradient gradient
-                            :shape shape}]
-             (case (:type gradient)
-               :linear [:> grad/linear-gradient props]
-               :radial [:> grad/radial-gradient props]))))
+                            :shape obj}]
+             (case (d/name (:type gradient))
```

Comment:
I'm not clearly understand this change, or the need to to execute the d/name for this.
The type should be always keyword

Reply:
Yes! but there was a bug that sometimes that was a string. This MR solves that bug but I was worried of old saved files. This was more for retro-compatibility than for anything else.

Reply:
There is a migration for it. I mean right now the file does not passes validation if type is string and the migration is very recent so there are no files with this bug I think.